src/Parser.php

```diff
@@ -119,65 +97,58 @@ public function __construct($stream, ListenerInterface $listener, string $lineEn
         if (!\is_resource($stream) || 'stream' !== get_resource_type($stream)) {
             throw new \InvalidArgumentException('Invalid stream provided');
         }
-
         $this->stream = $stream;
         $this->listener = $listener;
         $this->emitWhitespace = $emitWhitespace;
         $this->state = self::STATE_START_DOCUMENT;
         $this->bufferSize = $bufferSize;
         $this->lineEnding = $lineEnding;
     }
-
     public function parse(): void
     {
         $this->lineNumber = 1;
         $this->charNumber = 1;
         $eof = false;
-
         while (!feof($this->stream) && !$eof) {
             $pos = ftell($this->stream);
             $line = stream_get_line($this->stream, $this->bufferSize, $this->lineEnding);
-
+            $len = $this->getLengthOfString($line);
```

Comment:
IMO https://secure.php.net/manual/en/function.mb-strlen.php would be even more appropriate